### PR TITLE
feat: updates to use ui-shell web component

### LINKED ISSUE #1478: Integrate "Navbar" web component
### Feature description

We need to integrate the **Navbar** component from the "web-components" project.

The version of the **Navbar** component being integrated should be the most recent one and must include the new "Documentation" dropdown created in https://github.com/Qiskit/web-components/issues/100.

### Dependencies

Depends on:

- https://github.com/Qiskit/web-components/issues/100

### Additional context

The navigation bar is expected to look like this: https://www.figma.com/file/xc1jekL6jga6NyRH64k1mV/documentation-redesign?node-id=136%3A2941

Screenshot 2022-08-24 at 10 51 30

### PR DESCRIPTION
**Pull request by yoshiemuranaka**
## Changes
Updates `TheMenu` component to use the [ui-shell ](https://qiskit.github.io/web-components/?path=/story/ui-shell--default)web component. 

Resolves #1478 

## Implementation details
- Updates `@qiskit/web-components` to use latest release
- Removes unused MobileMenu component but kept TheMenu component as a wrapper for use in pug template
- Removes cypress test for the MobileMenu component
- Adds new CTA clicked event to links clicked from the ui-shell

## How to read this PR
Would appreciate extra 👀 on any code that can be removed with this PR

[Preview link](https://platypus-pr-1555.ecy0akwlcpw.us-south.codeengine.appdomain.cloud/course/introduction/why-quantum-computing)

## Screenshots
Screen Shot 2022-09-19 at 3 18 52 PM

### CODE REVIEW

**Comment by yoshiemuranaka** [comment]
CTA event properties added:

Screen Shot 2022-09-16 at 10 41 15 AM


https://user-images.githubusercontent.com/8097306/190701292-e3aa58d5-e7c4-4ef9-9dc8-e5d86877f9b0.mov

File: frontend/vue/mixins/menu.ts [hunk 1]
```diff
@@ -1,62 +0,0 @@
...
-import {
-  ORDERED_COMMUNITY_SUB_LINKS,
-  HOME_LINK,
-  COMMUNITY_LINK,
-  TUTORIALS_LINK,
-  DOCUMENTATION_LINK,
-  LEARN_LINK,
-  OVERVIEW_LINK,
-  NavLink
-} from '../../constants/menuLinks'
```
Comment (eddybrando) [inline frontend/vue/mixins/menu.ts:12]: I believe that most of the contents of the `frontend/constants/menuLinks.ts` file can now be removed.

Only `NavLink` seems to still be used.

**Review by eddybrando — CHANGES_REQUESTED** [review]
Awesome!

I just added some information about more code that can be removed.

Also, currently, the alignment of the header isn't the same as in prod, but I believe that this can be changed once the changes from https://github.com/Qiskit/web-components/pull/154 are released.

Untitled-1

File: frontend/vue/mixins/menu.ts (hunk 1 above)
Comment (yoshiemuranaka) [reply frontend/vue/mixins/menu.ts:12]: thanks @eddybrando ! resolved by https://github.com/Qiskit/platypus/pull/1555/commits/a5b40af0893161f8c61c80ec198144706f8c390d

File: frontend/constants/menuLinks.ts [hunk 2]
```diff
@@ -41,215 +41,4 @@ interface NavLink {
...
-  GITHUB_REPOSITORY,
-  COMMUNITY_LINK,
-  TUTORIALS_LINK,
-  DOCUMENTATION_LINK,
-  STAY_CONNECTED_LINKS,
-  LEARN_LINK,
-  OVERVIEW_LINK,
-  NavLink
-}
+export type { NavLink }
```
Comment (yoshiemuranaka) [inline frontend/constants/menuLinks.ts:44]: Added export `type` to resolve this lint error 👇 
Screen Shot 2022-09-19 at 9 29 29 AM

**Comment by yoshiemuranaka** [comment]
> Also, currently, the alignment of the header isn't the same as in prod, but I believe that this can be changed once the changes from [Qiskit/web-components#154](https://github.com/Qiskit/web-components/pull/154) are released.

Thanks @eddybrando for the memefied code review 😆 (great way to start my Monday)
The issue with the alignment is fixed with the newest [web-components release](https://github.com/Qiskit/web-components/pull/156), like you mentioned. 
Bumped the version here https://github.com/Qiskit/platypus/pull/1555/commits/b5b50b13a83b5de73da10474dbd0ec0c4b877161

Screen Shot 2022-09-19 at 9 25 44 AM

**Review by techtolentino — DISMISSED** [review]
This is great! 

👏🏼

**Review by vabarbosa — APPROVED** [review]
@yoshiemuranaka thank you for this. always great to see bunch of code removed/cleaned up.
i just have one comment (not sure if there was a discussion already) but it is more about the `ui-shell` behavior than this specific PR so i approved this PR anyway.

the account icon/link always goes to https://learn.qiskit.org/account/ even when viewing from the preview link or running locally. 

previously, it would go to the `/account/` for the current hostname so in the preview link it would be:

https://platypus-pr-1555.ecy0akwlcpw.us-south.codeengine.appdomain.cloud/account/

similarly running locally would go to

http://localhost:8080/account/

**Pull request merged**